homeassistant/components/onewire/sensor.py

```diff
@@ -426,11 +427,29 @@ def state(self) -> StateType:
         """Return the state of the entity."""
         return self._state
 
+    def get_temperature(self):
+        """Get the latest data from the device."""
+        attempts = 1
+        while True:
+            try:
+                return self._owsensor.get_temperature()
+            except UnsupportResponseException as ex:
+                _LOGGER.debug(
+                    "Cannot read from sensor %s (retry attempt %s): %s",
+                    self._device_file,
+                    attempts,
+                    ex,
+                )
+                time.sleep(0.2)
```

Comment:
We don't allow sleep in the executor thread pool. If we need to sleep it must be asyncio.sleep in a coroutine.